### PR TITLE
Fix PointerAddedEvent handling in LiveTestWidgetsFlutterBinding

### LINKED ISSUE #61100: `LiveTestWidgetsFlutterBinding` is not handling some `PointerEvent` correctly.
```Dart
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  testWidgets('Input PointerAddedEvent', (WidgetTester tester) async {
    await tester.pumpWidget(const MaterialApp(home: Text('Test')));
    await tester.pump();
    final TestGesture gesture = await tester.createGesture();
    await gesture.addPointer();
  });
}
```
https://dartpad.dev/902bb29d5ba7f0003593f6ada59cba32
(need to open console on dartpad to see the error message)

This test passes on a host machine, but if run it on a device, this assert will be violated: 

https://github.com/flutter/flutter/blob/f0e396710a8ad6af0942b2bcb805fd97d4f4765d/packages/flutter_test/lib/src/binding.dart#L1436

This is due to different treatment for the first `PointerAddedEvent` as the start of a gesture in `GestureBinding` (https://github.com/flutter/flutter/blob/f0e396710a8ad6af0942b2bcb805fd97d4f4765d/packages/flutter/lib/src/gestures/binding.dart#L174-L216) and in `LiveTestWidgetsFlutterBinding` (https://github.com/flutter/flutter/blob/f0e396710a8ad6af0942b2bcb805fd97d4f4765d/packages/flutter_test/lib/src/binding.dart#L1427-L1472)

Luckily we don't usually generate `PointerAddedEvent` using our tester APIs like `move` or `tap`, but when we want to simulate real world inputs, this event is there. 

-------
solving this issue is part of #60796, https://github.com/flutter/flutter/pull/60796/files#diff-1fc62c212be6fd3a5e10c523ea69cb28R1456

### PR DESCRIPTION
**Pull request by CareF**
## Description

See #61100

This was part of #60796

## Related Issues

Fixes #61100

## Tests

`packages/flutter_test/test/live_binding_test.dart`

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_test/test/live_binding_test.dart [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import 'package:flutter_test/flutter_test.dart';
+
+// This file is for testings that require a `LiveTestWidgetsFlutterBinding`
+void main() {
+  LiveTestWidgetsFlutterBinding();
+  testWidgets('Input PointerAddedEvent', (WidgetTester tester) async {
+    await tester.pumpWidget(const MaterialApp(home: Text('Test')));
+    await tester.pump();
+    final TestGesture gesture = await tester.createGesture();
+    await gesture.addPointer();
```
Comment (dnfield) [inline packages/flutter_test/test/live_binding_test.dart:15]: nit: consider adding a comment here or at the top of the test explaining that this mimics the start of a gesture as seen on a device.

**Review by dnfield — APPROVED** [review]
LGTM, thanks for the detailed explanation in the bug!

**Comment by CareF** [comment]
I realize one thing I need to solve before merge. Now I'm just treating `PointerAddedEvent` differently, but `PointerHoverEvent` and `PointerRemovedEvent` look like possibly having same issue. I don't have a situation/example to see when these events shows in real environment. I think I need confirm from someone more familiar with the motivation of these PointerEvents. @dnfield @liyuqian

**Comment by dnfield** [comment]
@CareF I think you'd have an easier time getting those events if you tested in web/chrome or on desktop (e.g. `flutter run -d macos` and then run the mouse cursor over the app).

It's fine to fix those in a separate PR if you don't have good test cases for them yet - or, if you can generate the test case, then you can go ahead and fix it here too.

**Review by liyuqian — APPROVED** [review]
LGTM. +1 to fix this first, and fix hover and other possible events later with more knowledge and reproducible test cases.

**Pull request merged**